scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -336,125 +346,162 @@ counts_mtx
-
-*Intro about DESeq2*
+It looks like we only got rid of one group.
 
-Before we identify differentially expressed genes, we will want to subset the pseudo-bulked `SingleCellExperiment` object to contain only the cell type that we are interested in comparing across the two subdiagnoses.
+The last step we want to do to prepare our dataset for DE is to subset the pseudo-bulked SCE object to contain only the cell type that we are interested in comparing across the two RMS subtypes.
+As mentioned previously, we want to are specifically interested in the `Tumor_Myoblast` cell type.
 
-```{r}
+```{r filter celltype}
+# create a new sce with only the tumor myoblasts
 tumor_myoblast_sce <- filtered_pseudobulked_sce[, filtered_pseudobulked_sce$celltype_broad == "Tumor_Myoblast"]
 ```
 
-This leaves us with 6 columns in our dataset that should all have `celltype_broad = Tumor_Myoblast`. 
+After filtering for our cell type of interest we should have a dataset with 6 columns, 1 for each group of `Tumor_Myoblast` cells in each of our 6 samples.
 
-```{r}
-table(tumor_myoblast_sce$celltype_broad)
-```
+### Perform differential expression with DESeq2
 
-Now we are ready to set up our `DESeq2` object. 
+Now we will use the `DESeq2` package to perform DE analysis on our pseudo-bulked SCE object.
+From this point, we can proceed in the same way we would if we had a bulk RNA-seq dataset with 6 samples.
+We will start with the unnormalized raw counts in the `counts` assay of the pseudo-bulked SCE and do the following with `DESeq2`:
 
-```{r}
-# deseq2 requires the coldata object as a separate input
-#  in data.frame format
+- Create a `DESeq` object
+- Normalize and log transform the counts data
+- Estimate dispersions and shrink estimates
+- Fit a negative binomial model and perform hypothesis testing using Wald statistics
+
+#### Create the `DESeq2` Object
+
+First we will create the `DESeq2` object.
+To create the object we will need the unnormalized counts matrix, the metadata associated with the samples that will help tell `DESeq2` which sample belongs to which group, and the design formula. 
+The design formula is used to indicate which columns of the metadata need to be considered in the DE comparison.
+For our experiment we are comparing gene expression between different RMS subtypes, stored in the `diagnosis_group` column of the `colData` in the pseudo-bulked SCE.
+
+```{r deseq object}
+# extract the metadata for each sample from sce as data.frame
 coldata_df <- as.data.frame(colData(tumor_myoblast_sce))
   
 # set up the deseq object 
 deseq_object <- DESeq2::DESeqDataSetFromMatrix(counts(tumor_myoblast_sce),
+                                               # metadata from coldata 
                                                colData = coldata_df,
+                                               # name of column to use for grouping samples 
                                                design = ~ diagnosis_group)
```

Comment:
We can actually simplify this, as we already have a summarized experiment (SCE is a SE)!
While this makes for a bit of an indirection, not having to convert the colData is a win in my book.

```suggestion
# set up the deseq object 
deseq_object <- DESeq2::DESeqDataSet(tumor_myoblast_sce,
                                     # formula for grouping samples 
                                     design = ~ diagnosis_group)
```

Reply:
>  as we already have a summarized experiment (SCE is a SE)!

I'd make sure to emphasize this point too during this step to explain why we can jump right in